src/Microsoft.Identity.Web/WebAppAuthenticationBuilderExtensions.cs

```diff
@@ -101,97 +101,99 @@ public static partial class WebAppAuthenticationBuilderExtensions
-                {
-                    options.Authority = AuthorityHelpers.BuildAuthority(microsoftIdentityOptions);
-                }
-
-                // This is a Microsoft identity platform Web app
-                options.Authority = AuthorityHelpers.EnsureAuthorityIsV2(options.Authority);
-
-                // B2C doesn't have preferred_username claims
-                if (microsoftIdentityOptions.IsB2C)
-                {
-                    options.TokenValidationParameters.NameClaimType = Constants.NameClaim;
-                }
-                else
-                {
-                    options.TokenValidationParameters.NameClaimType = Constants.PreferredUserName;
-                }
-
-                // If the developer registered an IssuerValidator, do not overwrite it
-                if (options.TokenValidationParameters.IssuerValidator == null)
-                {
-                    // If you want to restrict the users that can sign-in to several organizations
-                    // Set the tenant value in the appsettings.json file to 'organizations', and add the
-                    // issuers you want to accept to options.TokenValidationParameters.ValidIssuers collection
-                    options.TokenValidationParameters.IssuerValidator = AadIssuerValidator.GetIssuerValidator(options.Authority).Validate;
-                }
-
-                // Avoids having users being presented the select account dialog when they are already signed-in
-                // for instance when going through incremental consent
-                var redirectToIdpHandler = options.Events.OnRedirectToIdentityProvider;
-                options.Events.OnRedirectToIdentityProvider = async context =>
-                {
-                    var login = context.Properties.GetParameter<string>(OpenIdConnectParameterNames.LoginHint);
-                    if (!string.IsNullOrWhiteSpace(login))
-                    {
-                        context.ProtocolMessage.LoginHint = login;
-                        context.ProtocolMessage.DomainHint = context.Properties.GetParameter<string>(
-                            OpenIdConnectParameterNames.DomainHint);
+               MicrosoftIdentityOptions microsoftIdentityOptions = serviceProvider.GetRequiredService<IOptions<MicrosoftIdentityOptions>>().Value;
+               var b2cOidcHandlers = new AzureADB2COpenIDConnectEventHandlers(openIdConnectScheme, microsoftIdentityOptions);
+
+               options.SignInScheme = cookieScheme;
+
+               if (string.IsNullOrWhiteSpace(options.Authority))
+               {
+                   options.Authority = AuthorityHelpers.BuildAuthority(microsoftIdentityOptions);
+               }
+
+               // This is a Microsoft identity platform Web app
+               options.Authority = AuthorityHelpers.EnsureAuthorityIsV2(options.Authority);
+
+               options.TokenValidationParameters = options.TokenValidationParameters.Clone();
```

Comment:
This line is the only real change. The rest are just spacing/indent format fixes.